#### code repo for the manuscript:


#### _"Parallel dynamics of bacterial genome reduction across independent transitions to endosymbiosis"_


---


![alt text](https://upload.wikimedia.org/wikipedia/commons/7/7c/Journal.pbio.0050126.g001.png)


The establishment of symbiosis dramatically alters the evolution of the associated species, making symbiotic systems ideal models for studying the impact of lifestyle changes on genomes. 

Here, we focused on Enterobacterales, a large and ancient bacterial lineage that includes endosymbionts with diverse host associations, ranging from gut inhabitants to intracellular environments, and from horizontal to vertical transmission. Leveraging over two hundred genomes, along with cutting-edge single-copy gene concatenation and multi-copy gene family approaches, we inferred a robust phylogenetic framework that supports eleven independent transitions to endosymbiosis. 

Inferences on patterns of genome evolution confirm previous hypotheses about the processes underlying genome reduction: a substantial spike in gene loss always occurs simultaneously with the establishment of endosymbiosis, while a reduction in gene acquisition mechanisms is associated with the subsequent genome erosion. Furthermore, gene family loss frequencies were correlated across independent endosymbiotic clades; genes with more conserved functions and stronger constraints on sequence evolution are lost less frequently, suggesting that differences in gene essentiality and dispensability drive the observed parallelism. 

Our analyses contribute to the coming of age of the theory of genome evolution in symbiotic associations and provide novel insights into the importance of recombination as an opposing force against genome erosion.


---


In this repo there is a [html R markdown](https://htmlpreview.github.io/?https://raw.githubusercontent.com/for-giobbe/enterobacterales/main/markdown.html) which makes possible to  follow the manuscript analyses.

The same R markdown - alongside large files - can be found within a [zenodo]() repository and it can be used to fully and seamlessly reproduce the manuscript analyeses. To do so, unzipe the folder and use [knitr](https://yihui.org/knitr/) on the ```
.rmd``` file.


